src/equations/acoustic_perturbation_2d.jl

```diff
@@ -240,8 +240,8 @@ function boundary_condition_slip_wall(u_inner, normal_direction::AbstractVector,
     u_normal = normal[1] * u_inner[1] + normal[2] * u_inner[2]
 
     # create the "external" boundary solution state
-    u_boundary = SVector(u_inner[1] - 2.0 * u_normal * normal[1],
-                         u_inner[2] - 2.0 * u_normal * normal[2],
+    u_boundary = SVector(u_inner[1] - 2.0f0 * u_normal * normal[1],
+                         u_inner[2] - 2.0f0 * u_normal * normal[2],
```

Comment:
```suggestion
    u_boundary = SVector(u_inner[1] - 2 * u_normal * normal[1],
                         u_inner[2] - 2 * u_normal * normal[2],
```
Integers are even more generic than `Float32` and will result in the same code in the end.

Reply:
I considered that the original type is a float number, so I made changes to it as well. Do you want me to change all numbers like `2.0` back to a more generic type `Int`? Also, I don't know why you chose to use float number instead of an integer in the original code.

Reply:
This code was written some time ago. I think we were just less careful about generic types at that point. It would indeed be a great contribution (although a boring task) to generalize places like these to use integers.